src/chain/chain-supervision.cc

```diff
@@ -650,6 +650,37 @@ void AppendSupervision(const std::vector<const Supervision*> &input,
   }
 }
 
+bool AddWeightToFst(const fst::StdVectorFst &normalization_fst,
```

Comment:
What is this for? Can you just use the AddWeightToSupervisionFst?

Reply:
It uses fst::StdVectorFst as input, but AddWeightToSupervisionFst use Supervision object

Reply:
Ok. Modify AddWeightToSupervisionFst to call AddWeightToFst instead of duplicating the code.

Reply:
I am not sure, if Dan likes to modify original function!
It is better to keep it for now!